x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/SettingsConfig.java

```diff
@@ -128,6 +156,14 @@ public Integer getAlignCheckpointsForUpdate() {
         return alignCheckpoints;
     }
 
+    public Boolean getUsePit() {
+        return usePit != null ? (usePit > 0) || (usePit == DEFAULT_USE_PIT) : null;
+    }
```

Comment:
Why are we allowing `null` here on the server side? Either it is using `pit` or not.

I initially thought "oh this is for bwc", but you specifically set `this.usePit = DEFAULT_USE_PIT` when the supplied value is `null`.

Reply:
From what I can tell, the only time to allow `null` is on an update. And update never calls `getUsePit()`.